swap_meet/vendor.py

```diff
@@ -1,2 +1,219 @@
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+
+        other_vendor.inventory.append(my_item)
+
+        other_vendor.inventory.remove(their_item)
+        
+        self.inventory.append(their_item)
+    
+        return True
+        
+
+# Wave 4 - SJ
+# 7. Created an instance method swap_first_item takes one argument, other_vendor
+#    considers the first item in the instance's / self.inventory, 
+#    and the first item in the friend's/ other_vendor.inventory
+# 7a. if the inventory for either are 0, return False
+# 7b. otherwise add that first item from their inventory to mine
+# 7c. then remove that first item from their inventory 
+# 7d. likewise, add the first item from my inventory to their inventory
+# 7e. then remove my first item from my inventory
+# 7f. return True
+
+    def swap_first_item(self, other_vendor):
+        
+        if len(self.inventory) == 0  or len(other_vendor.inventory)== 0:
+            return False
+
+        self.inventory.append(other_vendor.inventory[0])
+        
+        other_vendor.inventory.remove(other_vendor.inventory[0])
+
+        other_vendor.inventory.append(self.inventory[0])
+
+        self.inventory.remove(self.inventory[0])
+                
+        return True
+
+
+    # Wave 6 - SJ
+    # Created instance methods: get_by_category
+    # get_by_category takes in one argument: a string, representing a category
+    # setting a new_list variable to contain the items we are searching for
+    # using a for loop to search through the inventory,
+    # if items in self.inventory match the category,
+    # return the list
+    # If there are no items in the inventory that match the category argument, 
+    # the method returns an empty list
+
+
+    def  get_by_category(self, category):
+        new_list = []
+        
+        for item in self.inventory:
+            if category == item.get_category():
+                new_list.append(item)
+                
+        return new_list
+    
+    
+    # Wave 6- SJ
+    # Created instance method: get_best_by_category  
+    # takes in one argument: a string that represents a category
+    # items_of_category is a list of items that match the given category
+    # loops through the inventory, for items w/ highest condition (5)
+    # AND the matches the category
+    # if there's only one item, return that item
+    # else, returns the item with the best condition in a certain category
+
+    
+    def  get_best_by_category(self, category):
+        
+        items_of_category = self.get_by_category(category)
+        
+        item_with_best_condition = None
+        
+        for item in items_of_category:
+            if item_with_best_condition == None:
+                item_with_best_condition = item
+            
+            elif item.condition > item_with_best_condition.condition:
+                item_with_best_condition = item
```

Comment:
This would be the perfect time to implement the concepts of **Higher Order Functions and lambda Functions**. We could use the Python `max` function with a `lambda` function like so: 
```python
def get_best_by_category(self, category):
        items = self.get_by_category(category)

        if not items: return None

        return max(items, key= lambda item: item.condition)
```

Reply:
so elegant